linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -442,117 +431,92 @@ private[emitter] object CoreJSLib {
-      locally {
-        val instance = varRef("instance")
-        defineFunction("noIsInstance", paramList(instance), {
-          Throw(New(TypeErrorRef,
-              str("Cannot call isInstance() on a Class representing a JS trait/object") :: Nil))
-        })
-      }
+      defineFunction1("noIsInstance") { instance =>
+        Throw(New(TypeErrorRef,
+            str("Cannot call isInstance() on a Class representing a JS trait/object") :: Nil))
+      },
 
-      locally {
-        val arrayClassData = varRef("arrayClassData")
-        val lengths = varRef("lengths")
-        val lengthIndex = varRef("lengthIndex")
-
-        // newArrayObject
-        defineFunction("newArrayObject", paramList(arrayClassData, lengths), {
-          Return(genCallHelper("newArrayObjectInternal", arrayClassData, lengths, int(0)))
-        })
+      defineFunction2("newArrayObject") { (arrayClassData, lengths) =>
+        Return(genCallHelper("newArrayObjectInternal", arrayClassData, lengths, int(0)))
+      },
 
-        // newArrayObjectInternal
+      defineFunction3("newArrayObjectInternal") { (arrayClassData, lengths, lengthIndex) =>
         val result = varRef("result")
         val subArrayClassData = varRef("subArrayClassData")
         val subLengthIndex = varRef("subLengthIndex")
         val underlying = varRef("underlying")
         val i = varRef("i")
-        defineFunction("newArrayObjectInternal",
-            paramList(arrayClassData, lengths, lengthIndex), {
-          Block(
-              const(result, New(arrayClassData DOT "constr",
-                  BracketSelect(lengths, lengthIndex) :: Nil)),
-              If(lengthIndex < (lengths.length - 1), {
-                Block(
-                    const(subArrayClassData, arrayClassData DOT "componentData"),
-                    const(subLengthIndex, lengthIndex + 1),
-                    const(underlying, result.u),
-                    For(let(i, 0), i < underlying.length, i.++, {
-                      BracketSelect(underlying, i) :=
-                        genCallHelper("newArrayObjectInternal", subArrayClassData, lengths, subLengthIndex)
-                    })
-                )
-              }, Skip()),
-              Return(result)
-          )
-        })
+
+        Block(
+          const(result, New(arrayClassData DOT "constr",
+              BracketSelect(lengths, lengthIndex) :: Nil)),
+          If(lengthIndex < (lengths.length - 1), Block(
+            const(subArrayClassData, arrayClassData DOT "componentData"),
+            const(subLengthIndex, lengthIndex + 1),
+            const(underlying, result.u),
+            For(let(i, 0), i < underlying.length, i.++, {
+              BracketSelect(underlying, i) :=
+                genCallHelper("newArrayObjectInternal", subArrayClassData, lengths, subLengthIndex)
+            })
+          )),
+          Return(result)
+        )
       }
+    )
 
+    private def defineObjectGetClassFunctions(): Tree = {
```

Comment:
This method was introduced, but it's never called. That probably explains the issue.